operator/controllers/credential_controller.go

```diff
@@ -62,24 +63,52 @@ func NewCredentialReconciler(helm helm.Client) *CredentialReconciler {
 // +kubebuilder:rbac:groups=servicebroker.metabroker.suse.com,resources=credentials,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=servicebroker.metabroker.suse.com,resources=credentials/status,verbs=get;update;patch
 
-const credentialReconcileTimeout = time.Second * 10
+const (
+	credentialReconcileTimeout   = 10 * time.Second
+	credentialUnbindingFinalizer = "unbinding.credentials.servicebroker.metabroker.suse.com"
+)
 
 // Reconcile reconciles a Credential resource.
 func (r *CredentialReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), credentialReconcileTimeout)
 	defer cancel()
 
-	_ = r.log.WithValues("credential", req.NamespacedName)
+	log := r.log.WithValues("credential", req.NamespacedName)
 
 	credential := &servicebrokerv1alpha1.Credential{}
 	if err := r.Get(ctx, req.NamespacedName, credential); err != nil {
 		if errors.IsNotFound(err) {
-			// TODO: The credential no longer exists; run any unbinding steps necessary.
+			log.Info("Credential deleted")
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, err
 	}
 
+	// The Credential is set for deletion, pending finalizers.
+	if !credential.ObjectMeta.DeletionTimestamp.IsZero() {
+		if stringutil.Contains(credential.ObjectMeta.Finalizers, credentialUnbindingFinalizer) {
```

Comment:
It may be useful to refactor this code into a separate function, for clarity.

Reply:
I didn't move to a single method, instead, I created some methods on the types: e570e3f.